lib/src/common/disposable.dart

```diff
@@ -256,9 +254,11 @@ class Disposable implements _Disposable, DisposableManagerV6, LeakFlagger {
     }
   }
 
-  final Set<Future> _awaitableFutures = new HashSet<Future>();
-  Completer<Null> _didDispose = new Completer<Null>();
-  final Set<_Disposable> _internalDisposables = new HashSet<_Disposable>();
+  final _awaitableFutures = new HashSet<Future>();
+  final _didDispose = new Completer<Null>();
+  LeakFlag _leakFlag;
+  final _internalDisposables = new HashSet<_Disposable>();
+  bool _isDisposalImminent = false;
```

Comment:
This name is fine, but thinking of other suggestions just for discussion:

  * wasDisposeRequested
  * hasDisposeBegun
  * didBeginDispose
  * isDisposeInProgress

Reply:
Yeah I was trying to find something that differentiated itself from `isDisposing`.. I wanted something that communicated the fact that disposal had been requested, but wasn't actually _in progress_ yet, since the instance is effectively unusable once disposal is in progress, but can be still be used up until that point.

Reply:
I kind of like the "requested" nomenclature

Reply:
If the poll is still open to suggestions I would submit `isAwaitingDisposal`.